src/pid.cpp

```diff
@@ -76,12 +82,30 @@ void Pid::initPid(double p, double i, double d, double i_max, double i_min, bool
 }
 
 void Pid::reset()
+{
+  reset(false);
+}
+
+void Pid::reset(bool save_iterm)
 {
   p_error_last_ = 0.0;
   p_error_ = 0.0;
-  i_error_ = 0.0;
   d_error_ = 0.0;
   cmd_ = 0.0;
+
+  // If last integral error is already zero, just return
+  if (std::abs(i_error_) < std::numeric_limits<double>::epsilon())
+  {
+    return;
+  }
```

Comment:
Do we need this check?

Reply:
maybe not? it does not take more effort to just zero it.

Reply:
in the earlier version we had to use the RT buffer here, that's why

Reply:
Got it